presidio_evaluator/evaluation/base_evaluator.py

```diff
@@ -335,45 +335,111 @@ def calculate_score(
+    def get_results_dataframe(
+        evaluation_results: List[EvaluationResult], entities: Optional[List[str]] = None
+    ) -> pd.DataFrame:
         """Return a DataFrame with the results of the evaluation.
 
-        Columns:
-        - sentence_id
-        - token text
-        - annotation
-        - prediction
-        - start_indices
+        :param evaluation_results: List of EvaluationResult objects containing the evaluation results.
+        :param entities: Optional list of entities to filter the results. If None, all entities are included.
+
+        :return: A pandas DataFrame with the following columns
+            - sentence_id
+            - token text
+            - annotation
+            - prediction
+            - start_indices
         """
 
         if not evaluation_results or not evaluation_results[0].tokens:
-            raise ValueError("The evaluation results should not be empty and must contain tokens. "
-                             "Ensure that the input samples have tokens.")
+            raise ValueError(
+                "The evaluation results should not be empty and must contain tokens. "
+                "Ensure that the input samples have tokens."
+            )
 
         rows_list = []
         for i, res in enumerate(evaluation_results):
             tokens = res.tokens
-            annotations = res.actual_tags
-            predictions = res.predicted_tags
+            annotations = BaseEvaluator._filter_entities(res.actual_tags, entities)
+            predictions = BaseEvaluator._filter_entities(res.predicted_tags, entities)
             start_indices = res.start_indices
             for j in range(len(tokens)):
-                rows_list.append({"sentence_id": i,
-                                  "token": tokens[j],
-                                  "annotation": annotations[j],
-                                  "prediction": predictions[j],
-                                  "start_indices": start_indices[j]})
+                rows_list.append(
+                    {
+                        "sentence_id": i,
+                        "token": tokens[j],
+                        "annotation": annotations[j],
+                        "prediction": predictions[j],
+                        "start_indices": start_indices[j],
+                    }
+                )
 
         results_df = pd.DataFrame(rows_list)
         return results_df
 
+    @staticmethod
+    def _filter_entities(
+        tags: List[str], entities: Optional[List[str]] = None
+    ) -> List[str]:
+        """
+        Filter the tags to only include the specified entities.
+        If entities is None, return all tags.
+        """
+        if entities is None:
+            return tags
+        return [tag if tag in entities else "O" for tag in tags]
 
     @staticmethod
-    def precision(tp: int, fp: int) -> float:
-        return tp / (tp + fp + 1e-100)
+    def precision(
```

Comment:
Extended the precision, recall, f metrics to work with either raw (tp, fp, fn) values or aggregated (tp, num_annotated, num_predicted)